chanbackup/single.go

```diff
@@ -125,12 +125,22 @@ func NewSingle(channel *channeldb.OpenChannel,
 	// go to recover, we'll present this in order to derive the private
 	// key.
 	_, shaChainPoint := btcec.PrivKeyFromBytes(btcec.S256(), b.Bytes())
+	
+	// If a channel is unconfirmed, the block height of the ShortChannelID
+	// is zero. This will lead to problems when trying to restore that
+	// channel as the spend notifier would get a height hint of zero.
+	// To work around that problem, we add the channel broadcast height
+	// to the channel ID so we can use that as height hint on restore.
+	chanId := lnwire.NewShortChanIDFromInt(channel.ShortChanID().ToUint64())
+	if chanId.BlockHeight == 0 {
+		chanId.BlockHeight = channel.FundingBroadcastHeight
```

Comment:
should we do something else signal that this is a pending channel and not the real channel id? we could set the high order bit, or make the assumption that channels won't be in coinbase outputs and detect them via `TxIndex == 0`.

i'm not super familiar with the restore logic, but is it possible that this invalid short channel id gets used in incorrect places after restoring? i'd be concerned that certain subsystems make assumptions about the validity of the short channel ids they receive

Reply:
I'm not at all familiar with the restore logic. That's why I originally favored the second option mentioned in #3690. But the second option means adding a new version to SCBs or switch to TLV encoding entirely. Both seemed like larger changes than using the short channel ID as a workaround. But I'm still not sure which way to go.

What you are proposing is that on restore, we extract the height hint from the short channel ID but then set the block height back to 0 if `TxIndex == 0` before handing it to the channel manager? That sounds like a nice workaround too. How likely is it that there would ever be a channel opening in a coinbase TX? You wouldn't be able to close the channel for 100 blocks, but other than that it's possible in theory even if I don't see the use case yet...

Reply:
> i'm not super familiar with the restore logic, but is it possible that this invalid short channel id gets used in incorrect places after restoring

The impact will be the channel that's inserted into the channel graph after recovery, and also the final close summary for a channel. IMO those two areas are mostly cosmetic, so they shouldn't affect any operational logic within the system. Closed channels are also looked up according to their channel point, which is stable, so they'll still be able to be loacted.

Reply:
what if someone has two pending channels with the same broadcast height? inserting into the graph will fail bc the edge already exists. my main concern is an invalid short chan id being used in the links

Reply:
@cfromknecht there shouldn't be any interaction in the link other than sending `ChannelReestablish`, and the links are indexed by their channel ID. Is there something else you had in mind?